mp4parse_capi/src/lib.rs

```diff
@@ -668,19 +668,28 @@ pub unsafe extern fn mp4parse_get_indice_table(parser: *mut mp4parse_parser, tra
         _ => {},
     }
 
+    // track.media_time is track based time unit.
+    let media_time = match (&track.media_time, &track.timescale) {
+        (&Some(t), &Some(s)) => {
+            let time_us = PresentationTime::new(t.0 as i64, s);
+            Some(time_us.to_us())
+        },
+        _ => None,
+    };
+
     // Find the track start offset time from 'elst'.
     // 'media_time' maps start time onward, 'empty_duration' adds time offset
     // before first frame is displayed.
     let offset_time =
-        match (&track.empty_duration, &track.media_time, &context.timescale) {
-            (&Some(empty_duration), &Some(media_time), &Some(scale)) => {
-                (empty_duration.0 as i64 - media_time.0 as i64) * scale.0 as i64
+        match (&track.empty_duration, &context.timescale, media_time) {
+            (&Some(empty_duration), &Some(scale), Some(media_time)) => {
+                (empty_duration.0 * scale.0) as i64 - media_time
             },
-            (&Some(empty_duration), _, &Some(scale)) => {
-                empty_duration.0 as i64 * scale.0 as i64
+            (&Some(empty_duration), &Some(scale), _) => {
+                (empty_duration.0 * scale.0) as i64
             },
-            (_, &Some(media_time), &Some(scale)) => {
-                (0 - media_time.0 as i64) * scale.0 as i64
+            (_, _, Some(media_time)) => {
+                media_time
```

Comment:
Yes...
Is that better to implement a convert function in TrackScaledTime/MediaScaledTime? So we can avoid this kind of error again?